CMakeLists.txt

```diff
@@ -323,7 +323,11 @@ if(LIB_INSTALL_DIR)
 endif()
 
 if (NOT EXISTS "${MBEDTLS_FRAMEWORK_DIR}/CMakeLists.txt")
-    message(FATAL_ERROR "${MBEDTLS_FRAMEWORK_DIR}/CMakeLists.txt not found. Run `git submodule update --init` from the source tree to fetch the submodule contents.")
+    if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git/")
+        message(FATAL_ERROR "${CMAKE_CURRENT_SOURCE_DIR}/.git/ not found. Please ensure you have downloaded a release version of Mbed TLS from GitHub.")
```

Comment:
I'm not sure about the ".git not found" part of this error message, from a user's perspective: if I go and download a release tarball, it won't contain a `.git` subdirectory either, so there seems to be a mismatch between reported problem and suggested solution.

Suggestion: start this message with ""${MBEDTLS_FRAMEWORK_DIR}/CMakeLists.txt not found (and this does not appear to be a git checkout)." - and add "(and this appears to be a git checkout)" to the other one.

Not sure about the 2nd sentence either: most likely this will happen because people _have_ downloaded an archive from github, except they've picked the wrong one. So I think the message should be more precise, like "Please ensure you have downloaded the right archive from the release page on github."